repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneRequestHandler.scala

```diff
@@ -110,6 +110,10 @@ class KeystoneRequestHandler(identityServiceUri: String, akkaServiceClient: Akka
         val roles = (json \ "access" \ "user" \ "roles").as[JsArray].value
           .map(jsRole => Role((jsRole \ "name").as[String], (jsRole \ "tenantId").asOpt[String]))
           .toVector
+        val authenticatedBy = (json \ "access" \ "token" \ "RAX-AUTH:authenticatedBy").asOpt[JsArray]
+          .map(_.value)
+          .map(_.map(_.as[String]))
+          .map(_.toVector)
```

Comment:
Interesting, I actually started by treating an empty option as equivalent to an empty list.

In the end, I left it as an option as a way of being explicit with the typing. That is, this field is a Rackspace extension, so it's fairly likely that it simply will not be present. Technically it could even be an empty array, although that is unlikely (and maybe prohibited by whatever specification details this field). This way, all of that information is carried along with this value. Existence is determined by the state of the option, while the content of the vector mirror the contents of the JSON array.

So I guess the short and sweet answer is that this leaves open the possibility to modify behavior based on any possible state.

Oh, and yes, it can totally be a list. Yay Scala upgrades! I used a Vector for the sake of consistency. I would kind of like to drop the conversion to a vector from all of these fields, but I wasn't sure that this was the right time to do it. However, I will totally do it if no one has any issues with that. I'll note that the case class we stick all of these values into only requires that they be `Seq`s (and, implicitly, that they be Serializable).